OpenSim/Simulation/Model/GeometryPath.cpp

```diff
@@ -136,28 +136,27 @@ generateDecorations(bool fixed, const ModelDisplayHints& hints,
         DefaultGeometry::drawPathPoint(mbix, lastPos, getColor(state), appendToThis);
 
     Vec3 pos;
-    for (int j = 1; j < pathPoints.getSize(); j++) {
-        PathPoint* point = pathPoints[j];
-        PathWrapPoint* mwp = dynamic_cast<PathWrapPoint*>(point);
-
-        if (mwp) {
-            // If the point is a PathWrapPoint and has surfacePoints,
-            // then this is the second of two tangent points for the
-            // wrap instance. So add the surface points to the display
-            // path before adding the second tangent point.
-            // Note: the first surface point is coincident with the
-            // first tangent point, so don't add it to the path.
-            Array<Vec3>& surfacePoints = mwp->getWrapPath();
-            const Transform& X_BG = mwp->getBody().getTransformInGround(state);
-            for (int j = 1; j<surfacePoints.getSize(); j++) {
-                pos = X_BG*surfacePoints.get(j);
+    int cnt = 0;
+    for (int i = 1; i < pathPoints.getSize(); ++i) {
+        PathPoint* point = pathPoints[i];
+        PathWrapPoint* pwp = dynamic_cast<PathWrapPoint*>(point);
+
+        if (pwp) {
+            // A PathWrapPoint provides points on the wrapping surface as Vec3s
+            Array<Vec3>& surfacePoints = pwp->getWrapPath();
+            // The surface points are expressed w.r.t. the wrap surface's body 
```

Comment:
Would prefer to see a more direct explanation of what the following line of code does. Perhaps something like "Get the body's transform w.r.t. ground because the surface points are expressed w.r.t. the wrap surface's body and we need to draw them in the ground frame".